source/agora/test/Base.d

```diff
@@ -356,12 +357,38 @@ public class FakeClockBanManager : BanManager
 public extern (C++) class TestNominator : Nominator
 {
 extern(D):
+    /// number of txs required for nomination
+    private ulong txs_to_nominate;
+
     ///
     public this (NetworkManager network, KeyPair key_pair, Ledger ledger,
-        TaskManager taskman)
+        TaskManager taskman, ulong txs_to_nominate)
     {
+        this.txs_to_nominate = txs_to_nominate;
         super(network, key_pair, ledger, taskman);
     }
+
+    /// Overrides the default behavior and changes nomination behavior based
+    /// on the TestConf 'txs_to_nominate' option
+    protected override bool prepareNominatingSet (out ConsensusData data) @safe
+    {
+        // if 0 take all txs, otherwise nominate exactly this many txs
+        this.ledger.prepareNominatingSet(data,
+            this.txs_to_nominate ? this.txs_to_nominate : ulong.max);
+        if (data.tx_set.length < this.txs_to_nominate)
+            return false;  // not enough txs
+
+        // defensive coding, same as base class
+        // (but may be overruled by derived classes)
+        if (auto msg = this.ledger.validateConsensusData(data))
+        {
+            // todo: need to log this
+            // log.fatal("tryNominate(): Invalid consensus data: {}. Data: {}", msg, data);
```

Comment:
Why is it a TODO ?

Reply:
The `log` is private and in the module scope of `Nominator`. So I wasn't sure what to do about this.

Reply:
Makes sense.